src/view_wrapper.tsx

```diff
@@ -65,7 +65,18 @@ export default class ViewWrapper<M> extends React.PureComponent<ViewWrapperProps
     this.setState(this.execContext.push(merge(state, pick(keys(state), childProps))));
   }
 
+  public componentWillReceiveProps(nextProps) {
+    const nextChildProps = nextProps.childProps;
+    const { container, childProps } = this.props;
+    if (!equals(nextChildProps, childProps) && container.accepts(Refresh)) {
```

Comment:
We're doing basically the same thing 3 times now. This should get factored out into a private helper that takes a message constructor and the props to pass in the message, and return a boolean to indicate if the message was dispatched.